src/tests/test_kube_api.py

```diff
@@ -406,15 +406,32 @@ def _late_binding_install(
     ) -> None:
         cls._bind_all(cluster_deployment, agents)
         cls._set_agent_cluster_install_machine_cidr(agent_cluster_install, nodes)
+        api_vip, _ = cls._get_vips(nodes)
 
         if len(nodes) == 1:
             cls.set_single_node_ip(cluster_deployment, nodes)
+            api_vip = get_ip_for_single_node(cluster_deployment, nodes.is_ipv4)
+
+        """
+        Add the API VIP DNS record to the assisted-service network
+        here so it can resolve by the time the host(s) start reclaim
+        """
+        if global_variables.reclaim_hosts:
+            nodes.controller.add_dns_host_to_network(
+                network_name="default", api_vip=api_vip, cluster_name=cluster_deployment.ref.name
+            )
 
         agent_cluster_install.wait_to_be_ready(ready=True)
         Agent.wait_for_agents_to_be_bound(agents)
         if not hold_installation:
             cls._wait_for_install(agent_cluster_install, agents)
 
+    @JunitTestCase()
+    def _reclaim_agents(self, agents: List["Agent"], cluster_deployment: ClusterDeployment, nodes: Nodes):
+        self._unbind_all(agents)
+        Agent.wait_for_agents_to_unbound(agents)
+        Agent.wait_for_agents_to_reclaim(agents)
```

Comment:
No need for `cluster_deployment` and `nodes` here (or I'm missing something?), also, you can make this method `classmethod`

```suggestion
    @classmethod
    @JunitTestCase()
    def _reclaim_agents(cls, agents: List["Agent"]):
        cls._unbind_all(agents)
        Agent.wait_for_agents_to_unbound(agents)
        Agent.wait_for_agents_to_reclaim(agents)
```

Reply:
Thanks for catching this! Modified!